glam/api/views.py

```diff
@@ -15,86 +15,32 @@
 
 
-@api_view(["POST"])
-def aggregations(request):
-    """
-    Fetches aggregation data.
-
-    Expects a JSON object in the body containing the query parameters, e.g.::
-
-        {
-            "query": {
-                "channel": "nightly",
-                "probe": "gc_ms",
-                "versions": ["70"],  # OR ["70", "69", "68"]
-                "aggregationLevel": "version"  # OR "build_id"
-            }
-        }
-
-    Returns a JSON object containing the histogram data and metadata, e.g.::
-
-        {
-            "response": [
-                {
-                    "data": [
-                        {
-                            "client_agg_type": "summed-histogram",
-                            "histogram": {
-                                "0": 0.0,
-                                "1": 1920.963,
-                                ...
-                            },
-                            "percentiles": {
-                                "0": 1.0,
-                                "10": 1.0259,
-                                ...
-                            },
-                            "total_users": 1604
-                        }
-                    ],
-                    "metadata": {
-                        "build_id": null,
-                        "channel": "nightly",
-                        "metric": "gc_ms",
-                        "metric_type": "histogram-exponential",
-                        "os": "Linux",
-                        "version": "70"
-                    }
-                }
-            ]
-        }
+def get_aggregations(**kwargs):
+    REQUIRED_QUERY_PARAMETERS = ["channel", "probe", "versions", "aggregation_level"]
 
-    """
     labels_cache = caches["probe-labels"]
     if labels_cache.get("__labels__") is None:
         Probe.populate_labels_cache()
 
-    REQUIRED_QUERY_PARAMETERS = ["channel", "probe", "versions", "aggregationLevel"]
-    body = request.data
-
-    if body is None or body.get("query") is None:
-        raise ValidationError("Unexpected JSON body")
-
-    q = body["query"]
-
-    if any([k not in q.keys() for k in REQUIRED_QUERY_PARAMETERS]):
+    if any([k not in kwargs.keys() for k in REQUIRED_QUERY_PARAMETERS]):
         # Figure out which query parameter is missing.
-        missing = set(REQUIRED_QUERY_PARAMETERS) - set(q.keys())
+        missing = set(REQUIRED_QUERY_PARAMETERS) - set(kwargs.keys())
         raise ValidationError(
             "Missing required query parameters: {}".format(", ".join(sorted(missing)))
         )
 
     dimensions = [
-        Q(metric=q.get("probe")),
-        Q(channel=CHANNEL_IDS[q.get("channel")]),
-        Q(version__in=map(str, q.get("versions"))),
-        Q(os=q.get("os")),
+        Q(metric=kwargs["probe"]),
+        Q(channel=CHANNEL_IDS[kwargs["channel"]]),
+        Q(version__in=map(str, kwargs["versions"])),
+        Q(os=kwargs["os"]),
     ]
+    aggregation_level = kwargs["aggregation_level"]
```

Comment:
Tests fail b/c this should be `kwargs["aggregationLevel"]`

Reply:
this actually isn't why the test fails. It fails because `q["aggregationLevel"]` doesn't exist I think, not `kwargs["aggregationLevel"]`. This worked before I pushed changes because I had the validation in the view and the `get_aggregations` function

Reply:
I'm going to revert back to camel case in `get_aggregations` because this is causing noise